### PR TITLE
Add branch provider

### PR DESCRIPTION
**Pull request by erikdrobne**

### CODE REVIEW

File: Analytical/Classes/Provider/BranchProvider.swift [hunk 1]
```diff
@@ -11,23 +11,25 @@ import Branch
...
     public func setup(with properties: Properties?) {
         
-        #if PRODUCTION
-            if UIApplication.environment == .testFlight {
-                Branch.setUseTestBranchKey(true)
-            }
-        #else
-            Branch.setUseTestBranchKey(true)
-        #endif
+        guard let shouldUseTestKey = properties?[BranchProvider.ShouldUseTestKey] as? Bool, let isDebugEnabled = properties?[BranchProvider.IsDebugEnabled] as? Bool else {
```
Comment (Legoless) [inline Analytical/Classes/Provider/BranchProvider.swift]: Not sure why those are required.. We need a more out of the box approach. By default `setup` should also call `initSession`. We can do this by setting a parameter such as `delayInitSession`, to allow the `metadata` setup. And `initSession` should not take `launchOptions` as parameter, but rather it should be stored internally from `setup`, if `delayInitSession` property is active.

**Review by Legoless — CHANGES_REQUESTED** [review]

File: Analytical/Classes/Provider/BranchProvider.swift (hunk 1 above)
Comment (erikdrobne) [reply Analytical/Classes/Provider/BranchProvider.swift]: Agree. I added the parameter for session initialization handling. I also refactored `launchOptions` handling.
We need the parameters `ShouldUseTestKey` & `IsDebugEnabled` to handle behavior based on the environment.

Comment (Legoless) [reply Analytical/Classes/Provider/BranchProvider.swift]: Yes, but they can have default values. :) Both should be false.

Comment (erikdrobne) [reply Analytical/Classes/Provider/BranchProvider.swift]: 👍

**Review by Legoless — APPROVED** [review]

**Pull request merged**